air/src/execution_step/boxed_value/stream_map.rs

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::stream::*;
+use super::ValueAggregate;
+use crate::execution_step::execution_context::stream_map_key::StreamMapKey;
+use crate::execution_step::ExecutionResult;
+use crate::JValue;
+
+use air_interpreter_data::GenerationIdx;
+use air_trace_handler::merger::ValueSource;
+use air_trace_handler::TraceHandler;
+
+use serde_json::json;
+use std::rc::Rc;
+
+fn from_key_value(key: StreamMapKey<'_>, value: &JValue) -> Rc<JValue> {
+    Rc::new(json!({ "key": key, "value": value }))
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct StreamMap {
+    stream: Stream,
+}
+
+impl StreamMap {
+    pub(crate) fn from_generations_count(previous_count: GenerationIdx, current_count: GenerationIdx) -> Self {
+        Self {
+            stream: Stream::from_generations_count(previous_count, current_count),
+        }
+    }
+
+    pub(crate) fn from_value(key: StreamMapKey<'_>, value: &ValueAggregate) -> Self {
+        let obj = from_key_value(key, value.get_result());
+        let value = ValueAggregate::new(
+            obj,
+            value.get_tetraplet(),
```

Comment:
One should be careful with that...
Let's discuss it tomorrow.

Reply:
Let's document that